frontend/server/src/Controllers/Contest.php

```diff
@@ -4855,6 +4877,17 @@ public static function apiUpdate(\OmegaUp\Request $r): array {
 
         $updateProblemset = true;
         $updateRequests = false;
+        
+        // Handle recommended flag - only available for admins and support team
+        if ($r->ensureOptionalBool('recommended') !== null) {
+            if (!\OmegaUp\Authorization::isSystemAdmin($r->identity) && 
+                !\OmegaUp\Authorization::isSupportTeamMember($r->identity)) {
+                throw new \OmegaUp\Exceptions\ForbiddenAccessException(
+                    'userNotAllowed'
+                );
+            }
+        }
```

Comment:
Maybe here it does make sense to throw the exception, although the nesting of conditions can be reduced:
```suggestion
        $recommended = $r->ensureOptionalBool('recommended');
        if (!is_null($recommended) && !\OmegaUp\Authorization::isSupportTeamMember($r->identity)) {
            throw new \OmegaUp\Exceptions\ForbiddenAccessException(
                'userNotAllowed'
            );
        }
```